modules/apps/layout/layout-reports-web/src/main/java/com/liferay/layout/reports/web/internal/portlet/action/GetLayoutReportsIssuesMVCResourceCommand.java

```diff
@@ -0,0 +1,167 @@
+import com.liferay.portal.kernel.model.Group;
+import com.liferay.portal.kernel.module.configuration.ConfigurationProvider;
+import com.liferay.portal.kernel.portlet.JSONPortletResponseUtil;
+import com.liferay.portal.kernel.portlet.bridges.mvc.BaseMVCResourceCommand;
+import com.liferay.portal.kernel.portlet.bridges.mvc.MVCResourceCommand;
+import com.liferay.portal.kernel.service.GroupLocalService;
+import com.liferay.portal.kernel.util.Http;
+import com.liferay.portal.kernel.util.ParamUtil;
+import com.liferay.portal.kernel.util.Portal;
+import com.liferay.portal.kernel.util.ResourceBundleUtil;
+
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import javax.portlet.ResourceRequest;
+import javax.portlet.ResourceResponse;
+
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Modified;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Cristina González
+ */
+@Component(
+	configurationPid = "com.liferay.layout.reports.web.internal.configuration.LayoutReportsPageSpeedConfiguration",
+	immediate = true,
+	property = {
+		"javax.portlet.name=" + LayoutReportsPortletKeys.LAYOUT_REPORTS,
+		"mvc.command.name=/layout_reports/get_layout_reports_issues"
+	},
+	service = MVCResourceCommand.class
+)
+public class GetLayoutReportsIssuesMVCResourceCommand
+	extends BaseMVCResourceCommand {
+
+	@Activate
+	@Modified
+	protected void activate(Map<String, Object> properties) {
+		_layoutReportsGooglePageSpeedConfigurationProvider =
+			new LayoutReportsGooglePageSpeedConfigurationProvider(
+				_configurationProvider,
+				ConfigurableUtil.createConfigurable(
+					LayoutReportsGooglePageSpeedConfiguration.class,
+					properties));
+	}
+
+	@Override
+	protected void doServeResource(
+			ResourceRequest resourceRequest, ResourceResponse resourceResponse)
+		throws Exception {
+
+		Locale locale = _portal.getLocale(resourceRequest);
+
+		try {
+			long groupId = ParamUtil.getLong(resourceRequest, "groupId");
+
+			Group group = _groupLocalService.fetchGroup(groupId);
+
+			if (group == null) {
+				_log.error("No API Key configured for the site " + groupId);
+
+				JSONPortletResponseUtil.writeJSON(
+					resourceRequest, resourceResponse,
+					JSONUtil.put(
+						"error",
+						"problems retrieving the API Key for the site " +
+							groupId));
+
+				return;
+			}
+
+			LayoutReportsDataProvider layoutReportsDataProvider =
+				new LayoutReportsDataProvider(
+					_layoutReportsGooglePageSpeedConfigurationProvider.
+						getApiKey(group));
+
+			String canonicalURL = ParamUtil.getString(
+				resourceRequest, "canonicalURL");
+
+			List<LayoutReportsIssue> layoutReportsIssues =
+				layoutReportsDataProvider.getLayoutReportsIssues(canonicalURL);
+
+			Stream<LayoutReportsIssue> stream = layoutReportsIssues.stream();
+
+			JSONPortletResponseUtil.writeJSON(
```

Comment:
If someone malicious discovers this endpoint, could they can get unlimited free API requests?

Should we check for some kind of permissions before calling the API?

Reply:
The problem with a permission, is that right now, you just need to have the view permission to see this statistics :(, probably we will add a new permission different that view/edit for it.

For now we have decided that the edit permission will be needed to call the API, but probably a more specific one will be created.

I will add the restriction for edit permission in LPS-119068, and the update this PR.

Thanks!!!

Thanks!!!